base/libgit2/types.jl

```diff
@@ -210,12 +204,24 @@ Matches the [`git_remote_callbacks`](https://libgit2.github.com/libgit2/#HEAD/ty
     payload::Ptr{Void}
 end
 
-function RemoteCallbacks(credentials_cb::Ptr{Void}, payload::Ref{<:Payload})
```

Comment:
I don't think this can be backported unless this non-kwarg method is preserved.

Reply:
I doubt it matters much, but we just leave this part out of the backport

Reply:
Then I guess the backport would have to be a separate PR against release-0.6 rather than a cherry pick from this commit?

Reply:
Also, fixing this bug requires changing the struct layout (and thus access), which is even more significantly breaking than this method. But like Keno pointed out, the GC was also completely different, so I can probably make a much more conservative fix there that assumes specific missing optimizations.